components/raftstore/src/store/worker/refresh_config.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+use crate::store::fsm::apply::{ApplyFsm, ControlFsm};
+use crate::store::fsm::store::StoreFsm;
+use crate::store::fsm::PeerFsm;
+use batch_system::{BatchRouter, Fsm, FsmTypes, HandlerBuilder, Poller, PoolState, Priority};
+use file_system::{set_io_type, IOType};
+use std::fmt::{self, Display, Formatter};
+use std::sync::Arc;
+use std::thread;
+use tikv_util::worker::Runnable;
+use tikv_util::{debug, error, info, safe_panic, thd_name};
+
+pub struct PoolController<N: Fsm, C: Fsm, H: HandlerBuilder<N, C>> {
+    pub router: BatchRouter<N, C>,
+    pub state: PoolState<N, C, H>,
+}
+
+impl<N, C, H> PoolController<N, C, H>
+where
+    N: Fsm,
+    C: Fsm,
+    H: HandlerBuilder<N, C>,
+{
+    #[allow(dead_code)]
+    pub fn new(router: BatchRouter<N, C>, state: PoolState<N, C, H>) -> Self {
+        PoolController { router, state }
+    }
+}
+
+impl<N, C, H> PoolController<N, C, H>
+where
+    N: Fsm + std::marker::Send + 'static,
+    C: Fsm + std::marker::Send + 'static,
+    H: HandlerBuilder<N, C>,
+{
+    pub fn decrease_by(&mut self, size: usize) {
+        for _ in 0..size {
+            if let Err(e) = self.state.fsm_sender.send(FsmTypes::Empty) {
+                error!(
+                    "failed to decrese thread pool";
+                    "decrease to" => size,
+                    "err" => %e,
+                );
+                return;
+            }
+        }
+    }
+
+    pub fn increase_by(&mut self, size: usize) {
+        let name_prefix = self.state.name_prefix.clone();
+        let mut workers = self.state.workers.lock().unwrap();
+        self.state.id_base += size;
```

Comment:
Perhaps we should update `id_base` after we use it to construct the new id `i + self.state.id_base`

Reply:
Fixed, thanks!